src/main/java/com/google/devtools/build/lib/packages/RuleClass.java

```diff
@@ -221,6 +221,19 @@ public int getPathSegment() {
     ENABLED,
     /** The rule should not use toolchain resolution. */
     DISABLED,
+    /**
+     * The rule instance uses toolchain resolution if it has a select().
+     *
+     * <p>This is for rules that don't intrinsically use toolchains but have select()s on {@link
+     * com.google.devtools.build.lib.rules.platform.ConstraintValue}, which are part of the build's
+     * platform. Such instances need to know what platform the build is targeting, which Bazel won't
+     * provide unless toolchain resolution is enabled.
+     *
+     * <p>Ultimately, we should remove this when <a
+     * href="https://github.com/bazelbuild/bazel/issues/12899#issuecomment-767759147}#12899</a>is
+     * addressed, so platforms are unconditionally provided for all rules.
+     */
+    HAS_SELECT,
```

Comment:
I had initially thought this was checking the Target (in ConfiguredTargetFunction) to see if any selects were available, not that we were hard-coding this for `alias` only. I'm fine with this choice, but add a comment here clarifying that this doesn't do any inspection, it's still set statically on a rule.

Reply:
Done.

I'm open to even following it the way you suggest.I just wanted to be conservative in how much we expand this functionality given the law of unpredictable consequences for large diffs.